lib/pages/portal/service/website_provider.dart

```diff
@@ -220,4 +248,26 @@ class WebsiteProvider with ChangeNotifier {
       return false;
     }
   }
+
+  void updateVisits(Website website) async {
+    website.numberOfVisits++;
+    List<String> websitesId =
+    PrefService.sharedPreferences.getStringList("websitesId");
```

Comment:
Use null-aware operators (see [this codelab](https://dart.dev/codelabs/dart-cheatsheet#null-aware-operators)) and remove the `if` checking for `null`s.

```suggestion
    PrefService.sharedPreferences.getStringList("websitesId") ?? [];
```